climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_method: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
```

Comment:
what does this here mean?

Reply:
This is a new thing to python3, type hinting (https://docs.python.org/3/library/typing.html). They use it a lot in `xarray`. It's why I have `prediction: xr.Dataset` in the signature. It just hints to developers what to expect for types.

I'm not sold that it's necessary in the function signatures, but I think it's nice.

Anyways, I really like this feature you're commenting on here. You can type hint expected variables. This doesn't do anything -- it doesn't initialize or create them. But for something like `Comparison` that has a bunch of subclasses, this tells the developer exactly which attributes are expected for a given subclass and what their type is. So you know when you make a new `Comparison` you need `comparison_method`, `hindcast_comparison`, etc.